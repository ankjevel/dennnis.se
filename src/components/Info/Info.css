.info {
  @apply px-6 py-4 mb-4 bg-red-400 min-w-full;
  h1 {
    @apply py-4;
  }

  @screen dark-mode {
    @apply bg-red-800 text-gray-400;
  }
}
