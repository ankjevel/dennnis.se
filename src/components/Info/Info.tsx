import React, { ReactElement } from 'react'

import css from './Info.css'

export const component = (props: {
  className?: string
  children: ReactElement
}) => {
  return (
    <header
      className={props.className ? `${css.info} ${props.className}` : css.info}
    >
      {props.children}
    </header>
  )
}

export default component
