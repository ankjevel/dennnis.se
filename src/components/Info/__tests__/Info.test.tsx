import React from 'react'
import { shallow } from 'enzyme'
import Info from '../Info'

describe('Info', () => {
  it('creates the component', () => {
    const component = shallow(
      <Info className="test">
        <h1>Hello</h1>
      </Info>
    )

    expect(component.html()).toMatchSnapshot()
  })

  it('contains the expected classnames', () => {
    const elem = shallow(
      <Info className="test">
        <h1>Hello</h1>
      </Info>
    )
    expect(elem.hasClass('info'))
    expect(elem.hasClass('test'))
  })
})
