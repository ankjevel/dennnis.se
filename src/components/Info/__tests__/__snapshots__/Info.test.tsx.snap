// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info creates the component 1`] = `"<header class=\\"info test\\"><h1>Hello</h1></header>"`;
